src/GitTfs.VsCommon/TfsHelper.Common.cs

```diff
@@ -667,24 +661,35 @@ protected ITfsChangeset BuildTfsChangeset(Changeset changeset, IGitTfsRemote rem
             return tfsChangeset;
         }
 
-        protected virtual bool HasWorkItems(Changeset changeset)
+        protected virtual IEnumerable<TfsWorkitem> GetWorkItems(Changeset changeset)
         {
-            // This method wraps changeset.WorkItems, because
-            // changeset.WorkItems might result to ConnectionException: TF26175: Team Foundation Core Services attribute 'AttachmentServerUrl' not found.
-            // in this case assume that it is initialized to null
-            // NB: in VS2011 a new property appeared (AssociatedWorkItems), which works correctly
-            WorkItem[] result = null;
+            var workItems = changeset.AssociatedWorkItems.Select(wi => new TfsWorkitem
```

Comment:
Can we have a comment in this method explaining why we are first using the `AssociatedWorkItems` and then
the `WorkItems`?
I already have spent several minutes on the `Join` below to understand that it will only return the WorkItems which are in both sets (the intersection). I fear that in  half a year, I would be completely lost when looking at the code again.